operator/apis/metrics/v1alpha1/keptnmetric_types.go

```diff
@@ -80,3 +85,10 @@ func init() {
 func (s KeptnMetric) IsStatusSet() bool {
 	return s.Status.Value != ""
 }
+
+func checkAllowedProvider(provider string) error {
```

Comment:
is this method necessary ? We already have a helper method on `keptnmetric_webhook.go` -> we can move the logic there